django-server/server/api/models.py

```diff
@@ -65,16 +65,49 @@ class Meta:
         verbose_name_plural = _('cameras')
 
 
-class CameraEvent(models.Model):
-    class EventType(models.IntegerChoices):
-        MONITORING_STARTED = 0, _('monitoring_started')
-        MONITORING_ENDED = 1, _('monitoring_ended')
-        PERSON_ENTERED = 2, _('person_entered')
-        PERSON_LEFT = 3, _('person_left')
-        OCCUPANCY_OVERRIDE = 4, _('occupancy_override')
+class CameraEventType(models.IntegerChoices):
+    MONITORING_STARTED = 0, _('monitoring_started')
+    MONITORING_ENDED = 1, _('monitoring_ended')
+    PERSON_ENTERED = 2, _('person_entered')
+    PERSON_LEFT = 3, _('person_left')
+    OCCUPANCY_OVERRIDE = 4, _('occupancy_override')
+
+
+# For enforcing event_type.camera is (non-)null 
+def get_q_event_camera_nullability():
+    null_cam_event_types = [
+        CameraEventType.OCCUPANCY_OVERRIDE
+    ]
+    nonnull_cam_event_types = [
+        CameraEventType.MONITORING_STARTED,
+        CameraEventType.MONITORING_ENDED,
+        CameraEventType.PERSON_ENTERED,
+        CameraEventType.PERSON_LEFT,
+    ]
+    return Q(event_type__in=null_cam_event_types, camera__isnull=True) | Q(
+        event_type__in=nonnull_cam_event_types, camera__isnull=False
+    )
+
+
+# For enforcing event_type.event_value is (non-)null 
+def get_q_event_value_nullability():
+    null_val_event_types = [
+        CameraEventType.MONITORING_STARTED,
+        CameraEventType.MONITORING_ENDED,
+        CameraEventType.PERSON_ENTERED,
+        CameraEventType.PERSON_LEFT,
+    ]
+    nonnull_val_event_types = [
+        CameraEventType.OCCUPANCY_OVERRIDE
+    ]
+    return Q(event_type__in=null_val_event_types, event_value__isnull=True) | Q(
+        event_type__in=nonnull_val_event_types, event_value__isnull=False
+    )
```

Comment:
is it the preffered way of formating?

Reply:
I ran it these lines through `black` formatter.